### PR TITLE
refine savings plan estimates

### PR DESCRIPTION
**Pull request by AjayTripathy**
For time periods where savings plans expired, ran out of allocated credits, or had partial usages covered, we were underestimating the hourly cost. My proposal is we instead charge the savings plan rate to each node. This is still likely an underestimate, in the sense that the real value is the partial coverage in the "effective cost" column + the regular line item usage. 

For perfect accuracy you need to wait for reconciliation data anyway-- my suggestion is just to improve the estimate here for now, instead of trying to perfectly calculate an hourly cost w/ all things like amortized initial savings plan fee and amortized monthly recurring savings plan fee etc.

### CODE REVIEW

**Comment by dwbrown2** [comment]
Can you share the difference between `savings_plan_savings_plan_effective_cost` and `savings_plan_savings_plan_rate`? I would expect the latter to be a better option and so it's not clear why this PR is an improvement.

**Comment by AjayTripathy** [comment]
"savings_plan_savings_plan_rate": The hourly rate of the node under the savings plan.
"savings_plan_effective_cost": The amount actually billed against the savings plan. 

In cases where you exceed your savings plan, savings_plan_effective cost is low, as there is a "remainder" line item that appears elsewhere in the CUR. We could try to search the CUR for this line when effective cost < rate, but given that we do this in the price reconciliation pipeline I think this is overkill here.

**Comment by dwbrown2** [comment]
SGTM, thanks for sharing

Webb
Kubecost, Co-founder

ᐧ

File: pkg/cloud/awsprovider.go [hunk 1]
```diff
@@ -1752,8 +1756,20 @@ func (a *AWS) GetSavingsPlanDataFromAthena() error {
...
 		}
 		a.SavingsPlanDataLock.Unlock()
-	} else {
-		klog.Infof("No savings plan applied instance data found")
+		return true
+	}
+
+	query := fmt.Sprintf(q, cfg.AthenaTable, start, end)
+
+	klog.V(3).Infof("Running Query: %s", query)
```
Comment (nikovacevic) [inline pkg/cloud/awsprovider.go:1764]: Noisy log?

**Review by nikovacevic — APPROVED** [review]
LGTM

File: pkg/cloud/awsprovider.go (hunk 1 above)
Comment (AjayTripathy) [reply pkg/cloud/awsprovider.go:1764]: It doesn't happen that often and has been useful in debugging so far.

**Pull request merged**